File: Modules/ServerTypes/Announcement/AnnouncementForm.ts
import { StringEditor, MultipleImageUploadEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface AnnouncementForm {
    AnnouncerID: StringEditor;
    AnnouncerName: StringEditor;
    AnnouncementContent: StringEditor;
    UploadDocument: MultipleImageUploadEditor;
}

export class AnnouncementForm extends PrefixedContext {
    static formKey = 'Announcement.Announcement';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AnnouncementForm.init)  {
            AnnouncementForm.init = true;

            var w0 = StringEditor;
            var w1 = MultipleImageUploadEditor;

            initFormType(AnnouncementForm, [
                'AnnouncerID', w0,
                'AnnouncerName', w0,
                'AnnouncementContent', w0,
                'UploadDocument', w1
            ]);
        }
    }
}

File: Modules/HumanResource/Announcement/Announcement/AnnouncementForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.Announcement.Forms;

[FormScript("Announcement.Announcement")]
[BasedOnRow(typeof(AnnouncementRow), CheckNames = true)]
public class AnnouncementForm
{
    [HalfWidth]
    public string AnnouncerID { get; set; }
    [HalfWidth]
    public string AnnouncerName { get; set; }

    public string AnnouncementContent { get; set; }
    public string UploadDocument { get; set; }

}